spec/library_spec.rb

```diff
@@ -0,0 +1,34 @@
+require 'yaml'
+require './lib/library.rb'
+
+#collection = YAML.load_file('./lib/booksdb.yml')
+
+describe Library do
+  let(:patron) {double('Patron')}
+
+  it 'should have the list of books' do
+    expect(subject.bookrack).not_to be nil
+  end
+
+  it 'should check if the book is available' do
+     expect(subject.bookrack[0][:available]).to be true
+  end
+
+  it 'should return error if book not available' do
+     expect{subject.check_book_availability(100)}.to raise_error('Book not available')
+  end
+
+  describe "Checked-out process:" do
+  before {subject.check_book_availability(0)}
+    it 'can checkout book to patron and update the bookrack' do
+      expect(subject.bookrack[0][:available]).to be false
+    end
+  end
```

Comment:
This method does *not* test that you are checking out a book. Call the method that checks a book out to a patron.